src/toil/job.py

```diff
@@ -1176,13 +1225,13 @@ def readGlobalFile(self, fileStoreID, userPath=None, cache=True, mutable=None):
                 # FileStoreID exists.  If it does, the wait and periodically check for the removal
                 # of the file and the addition of the completed download into cache of the file by
                 # the other job. Then we link to it.
-                elif fileIsLocal and os.path.exists(harbingerFileName):
+                elif fileIsLocal and harbingerFile.exists():
```

Comment:
It messes up the conditionals then. The else in the if harbingerFile.exists is the same as the else in the outer block